### PR TITLE
Allow several patterns in select helpers

### PR DESCRIPTION
**Pull request by etiennebacher**
Mentioned in the discussion in #258. Examples:
``` r
library(datawizard)

find_columns(iris, starts_with("Petal", "Sep"))
#> [1] "Sepal.Length" "Sepal.Width"  "Petal.Length" "Petal.Width"

find_columns(iris, ends_with("Length", "Width"))
#> [1] "Sepal.Length" "Sepal.Width"  "Petal.Length" "Petal.Width"

find_columns(iris, contains("et", "idt"))
#> [1] "Sepal.Width"  "Petal.Length" "Petal.Width"
```

<sup>Created on 2022-09-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

File: R/select_helpers.R [hunk 1]
```diff
@@ -300,3 +303,18 @@
...
+
+
+# extract variable names from deparsed select helpers
+# e.g "starts_with(\"Sep\", \"Petal\")" -> c("Sep", "Petal")
+.extract_vars_from <- function(x, select_helper) {
+  tmp <- gsub(paste0(select_helper, "\\(\"(.*)\"\\)"), "\\1", x)
+  tmp <- gsub("'", "", tmp)
+  tmp <- gsub('"', "", tmp)
+  tmp <- strsplit(tmp, ",")[[1]]
+  tmp <- trimws(tmp)
```
Comment (strengejacke) [inline R/select_helpers.R:315]: Can you use `insight::trim_ws()` here?

File: R/select_helpers.R [hunk 2]
```diff
@@ -300,3 +303,18 @@
     }
   }
 }
+
+
+# extract variable names from deparsed select helpers
+# e.g "starts_with(\"Sep\", \"Petal\")" -> c("Sep", "Petal")
+.extract_vars_from <- function(x, select_helper) {
+  tmp <- gsub(paste0(select_helper, "\\(\"(.*)\"\\)"), "\\1", x)
+  tmp <- gsub("'", "", tmp)
```
Comment (strengejacke) [inline R/select_helpers.R:312]: Can we add `fixed = TRUE` here? (for both)

**Review by strengejacke — CHANGES_REQUESTED** [review]

**Comment by etiennebacher** [comment]
Done, failing tests still due to XML on latest ubuntu

**Comment by strengejacke** [comment]
The lintr-comments regarding "sapply in undesired" can be changes like this:

``` r
x <- list(a = 1, b = "a", c = factor("d"), d = 2)
vapply(x, is.numeric, FUN.VALUE = TRUE)
#>     a     b     c     d 
#>  TRUE FALSE FALSE  TRUE
sapply(x, is.numeric)
#>     a     b     c     d 
#>  TRUE FALSE FALSE  TRUE
```

**Comment by strengejacke** [comment]
An `vapply` is faster, actually:

``` r
x <- list(a = 1, b = "a", c = factor("d"), d = 2)

microbenchmark::microbenchmark(
  vapply(x, is.numeric, FUN.VALUE = TRUE),
  sapply(x, is.numeric),
  times = 1000
)
#> Unit: microseconds
#>                                     expr  min   lq    mean median   uq   max
#>  vapply(x, is.numeric, FUN.VALUE = TRUE)  3.5  3.8  4.2199    4.2  4.4  27.7
#>                    sapply(x, is.numeric) 15.7 16.4 17.2389   16.6 16.9 109.7
```

<sup>Created on 2022-09-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Review by IndrajeetPatil — APPROVED** [review]
Thanks for the awesome work, @etiennebacher! 🙌

**Pull request merged**